Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -2009,8 +2011,27 @@ static bool isContextMenuEvent(const WebMouseEvent& event)
 #endif // ENABLE(CONTEXT_MENUS)
 }
 
-bool UnifiedPDFPlugin::handleKeyboardEvent(const WebKeyboardEvent&)
+bool UnifiedPDFPlugin::handleKeyboardEvent(const WebKeyboardEvent& event)
 {
+#if PLATFORM(MAC)
+    auto& commands = event.commands();
+    if (commands.size() != 1)
+        return false;
+    auto commandName = commands[0].commandName;
+
+    CheckedPtr keyboardAnimator = scrollAnimator().keyboardScrollingAnimator();
+
+    if (commandName == "scrollToBeginningOfDocument:"_s) {
+        keyboardAnimator->beginKeyboardScrollGesture(ScrollDirection::ScrollUp, ScrollGranularity::Document, false);
```

Comment:
Should we return the result of `KeyboardScrollingAnimator::beginKeyboardScrollGesture`?

Reply:
mayyyybe!